lib/protocol/collection/item_collection.ts

```diff
@@ -117,23 +117,46 @@ export class ItemCollection extends MutableCollection<SNItem> {
       /** Filtered content type map */
       const filteredCTMap = this.filteredMap[contentType]!;
       const sortedElements = this.sortedMap[contentType]!;
+
+      const previousIndex = filteredCTMap[element.uuid];
+      const previousElement = !isNullOrUndefined(previousIndex) ? sortedElements[previousIndex] : undefined;
+      /**  New value errorDecrypting  | previous value errorDecrypting | action(s)
+       * 1.           true            |           undefined            |    sort + change, continue loop
+       * 2.           false           |           undefined            |    not going inside clause
+       * 3.           true            |           true                 |    save, continue loop
+       * 4.           false           |           true                 |    sort + save
+       * 5.           true            |           false                |    sort + save, continue loop
+       * 6.           false           |           false                |    not going inside clause
+       */
+      if (element.errorDecrypting || previousElement?.errorDecrypting) {
+        if (element.errorDecrypting !== previousElement?.errorDecrypting) {
+          /** If errorDecrypting property changes a re-sort is necessary. */
+         typesNeedingResort.add(contentType);
+        }
+         /** We always save the value, but we don't continue in the for loop
+          * in case the errorDecrypting is false in the new value as we want filters to check it.  */
+        sortedElements[previousIndex] = element;
+        if (!element.errorDecrypting) {
+          continue;
```

Comment:
Wait why would you continue if the element is not error decrypting? If it is valid, then we should proceed to filtering. If it is invalid, we should skip filtering.

However, come to think about it, even if an item is invalid, we don't want to skip filtering, because the [filter is how clients choose](https://github.com/standardnotes/web/blob/004/app/assets/javascripts/views/notes/notes_view.ts#L320) which notes to display based on active tag. In this case it would seem like we would always display errorDecrypting items for every tag a user selects.

I'm thinking that perhaps clients should be the ones to do safety check for errorDecrypting. Perhaps the client wants to display a list of items that are errorDecrypting, sorted by created_at, or something like that? In which case, I think we could scrap this PR in its entirety.

Reply:
Yeah, that might be true. I removed most of the logic, but I left safe emergency safety checks and the test so IMO we should merge this.